pkg/agent/handler/accessrequest.go

```diff
@@ -122,6 +159,38 @@ func (h *accessRequestHandler) getManagedApp(ar *mv1.AccessRequest) (*v1.Resourc
 	return h.client.GetResource(url)
 }
 
+func (h *accessRequestHandler) addSubscription(ar *mv1.AccessRequest) {
+	// TODO - Use subscription reference subresource on AccessRequest instead of custom subresource
+	// once controller starts to populate it.
+	ars := ar.GetSubResource("x-marketplace-subscription")
```

Comment:
If this will be a common subresource we will be using, can you please add it as a const to the definitions package

Reply:
This is temporary change as the asset-controller is not populating the subscription references in the AccessRequest resource yet. Once that is done, we will remove this custom subresource. Added a TODO for it.